src/diffusers/models/prior_transformer.py

```diff
@@ -92,18 +118,27 @@ def __init__(
             ]
         )
 
+        if self.config.out_dim is not None:
+            self.norm_in = nn.LayerNorm(inner_dim)
+        else:
+            self.norm_in = None
+
         self.norm_out = nn.LayerNorm(inner_dim)
-        self.proj_to_clip_embeddings = nn.Linear(inner_dim, embedding_dim)
+
+        self.proj_to_clip_embeddings = nn.Linear(inner_dim, out_dim)
 
         causal_attention_mask = torch.full(
             [num_embeddings + additional_embeddings, num_embeddings + additional_embeddings], -10000.0
         )
         causal_attention_mask.triu_(1)
         causal_attention_mask = causal_attention_mask[None, ...]
         self.register_buffer("causal_attention_mask", causal_attention_mask, persistent=False)
-
-        self.clip_mean = nn.Parameter(torch.zeros(1, embedding_dim))
-        self.clip_std = nn.Parameter(torch.zeros(1, embedding_dim))
+        if self.config.out_dim is None:
+            self.clip_mean = nn.Parameter(torch.zeros(1, out_dim))
+            self.clip_std = nn.Parameter(torch.zeros(1, out_dim))
+        else:
+            self.clip_mean = None
+            self.clip_std = None
```

Comment:
```suggestion
        if self.config.out_dim is None:
            self.clip_mean = nn.Parameter(torch.zeros(1, out_dim))
            self.clip_std = nn.Parameter(torch.zeros(1, out_dim))
        else:
            self.clip_mean = None
            self.clip_std = None
```

This logic doesn't make sense to me. Ok for me to add a new config parameter if we have to and it looks like we have to here. Just make sure to add a parameter of type string to make it extendable for the future

Reply:
so `clip_mean` and `clip_std` is only used in `post_process_latents`, which is called separately in unclip pipelines. shap-e checkpoint does not contain these so I decided to handle the missing keys in the conversion script instead of adding a new config here